lib/Dialect/Calyx/CalyxOps.cpp

```diff
@@ -2747,10 +2783,12 @@ LogicalResult InvokeOp::verify() {
     return emitOpError() << "with instance '@" << callee
                          << "', which does not exist.";
   // The argument list of invoke is empty.
-  if (getInputs().empty())
+  if (getInputs().empty() && getRefCellsMap().empty()) {
     return emitOpError() << "'@" << callee
                          << "' has zero input and output port connections; "
+                            " and has not passing-by-reference cells; "
```

Comment:
I'm not totally following the logic here, but I also don't know why an invoke with empty argument is illegal either :)